### ISSUE TITLE
Refactoring dimensionality functionalities

### ISSUE THREAD

**Issue by utf** [materialsproject/pymatgen#1322]
Currently there are two dimensionality finders in pymatgen, one in `pymatgen.analysis.find_dimension`, the other in `pymatgen.analysis.structure_analyzer`. I am about to submit a PR for a third dimensionality finding routine which is faster and more reliable than both the existing methods. 

It seems like we should compile these together in a module and rename the methods to make it clear what they are doing.

For example the method in `pymatgen.analysis.structure_analyzer` is based on cleaving surfaces. So could be called `get_dimensionality_from_available_surfaces`.

The method in  `pymatgen.analysis.find_dimension` is based on building a supercell and counting the number of clusters so could be called `get_dimensionality_from_num_clusters`.

My method takes a StructureGraph object so could be called `get_dimensionality_from_bonded_structure()`.

To be honest these names aren't great but are better than the existing situation.

Obviously, any refactor will include deprecation warnings.

@shyuep @mkhorton Do you have any views on this?

**Comment by mkhorton** [comment]
Definitely should be combined into a common place. Perhaps all three in `pymatgen.analysis.dimensionality`? I could see a `Structure.get_dimension(algorithm='...')` convenience method maybe.

I believe the author of `pymatgen.analysis.find_dimension` is currently re-factoring their implementation for speed and reliability, but we can still move it.

Since all three different algorithms each are implemented based on an existing paper, I have a mild preference for renaming them to include the first author name. This is similar to what we do for the NearNeighbor classes.

**Comment by mkhorton** [comment]
My big concern generally is discoverability. I think having it in StructureAnalyzer is probably the most sensible place, but I think it's easy for this to get missed.

**Comment by utf** [comment]
Agree that `pymatgen.analysis.dimensionality` seems like a logical place.

Ok, so I could rename the three methods to just:
```python
get_dimensionality_larsen(...)
get_dimensionality_gorai(...)
get_dimensionality_cheon(...)
```

**Comment by utf** [comment]
This is fixed in #1323
